File: stack/739_daily_temperatures.py
class Solution:
    # O(n) time | O(n) space - where n is temperatures.length
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        ans = [0] * len(temperatures)
        stack = []
        for idx, temp in enumerate(temperatures):
            while stack and temperatures[stack[-1]] < temp:
                prevIdx = stack.pop()
                ans[prevIdx] =  idx - prevIdx
            stack.append(idx)
        return ans

File: stack/739_daily_temperatures.go
// O(n) time | O(n) space - where n is temperatures.length
func dailyTemperatures(temperatures []int) []int {
    ans := make([]int, len(temperatures))
    for i := range ans {
        ans[i] = 0
    }

    stack := []int{}
    for idx, temp := range temperatures {
        for len(stack) > 0 && temp > temperatures[stack[len(stack) - 1]] {
            prevIdx := stack[len(stack) - 1]
            stack = stack[:len(stack) - 1]
            ans[prevIdx] = idx - prevIdx
        }
        stack = append(stack, idx)
    }
    return ans
}
